students/danwoj/Session04/Prototypes/mailroom_dict.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3.6
+
+# This creates a statis donor table that does not retain values after this program runs
+
+donor_fnames = ['Bill', 'Melvin', 'Daphnie', 'Chauncey', 'Frieda']
+donor_lnames = ['Gates', 'Smith', 'Jones', 'Doe', 'Whatever']
+donations = [[200, 500], [2000.00, 3000.50, 3000.00], [1500.00, 500.25], [400.00], [2000.12, 3000, 4000]]
+
+keys = ['f_name', 'l_name', 'don_amt', 'don_num']
+donor_dict = dict.fromkeys(keys, None)
+donor_dict['f_name'] = donor_fnames
+donor_dict['l_name'] = donor_lnames
+donor_dict['don_amt'] = donations
+donor_dict['don_num'] = []
+
+for i in donor_dict['don_amt']:
+	donor_dict['don_num'].append(len(i))
+
+# This is the function that prints off the donor report
+def print_report():
+	print('\n#####################  Donor Report ####################\n')
+	print('Donor Name      | Total Given | Num Gifts | Average Gift')
+	print('--------------------------------------------------------')
+	i = 0
+	# This loop goes through each donor record in the dataset
+	for i in range (len(donor_names)):
+		j = 0
+		tot_don = 0
+		avg_don = 0
+		# This loop creates the total and average donation values for each donor
+		for j in range (len(donations[i])):
+			tot_don = tot_don + donations[i][j]
+			avg_don = avg_don + donations[i][j]
+		avg_don = avg_don / (j+1)
+		# This prints off each donor's row in the report
+		print(donor_names[i], ' '*(14-len(donor_names[i])), '| $', '%.2f' % tot_don, ' '*(8-len('%.2f' % tot_don)), '|', j+1, ' '*(8-len(str(j+1))), '| $', '%.2f' % avg_don)
+	return True
+
+# This function prints out the 'thank you' message for a new donation
+def ty_message(dn, da):
+	print('\nDear', dn, ',\n\n'
+	'Thank you so much for your generous donation of $', '%.2f' % da,'. This contribution is so important to the work our organization does and we express our deepest appreciation of your support to our important cause.')
+	return True
+
+# This is the function that prints off a thank you message for new donations
+def thank_you():
+	# Prints off basis input menu
+	tym_value = int(input('\n#####################\n#  Send Thank You   #\n#####################\n'
+		'1: List Donors\n'
+		'2: Enter Donation\n'
+		'3: Go Back\n'
+		'Enter your number choice: '))
+	# Creates a donor list if the user selects that option from the menu
+	if tym_value == 1:
+		i = 0
+		print('\nDONOR LIST:')
+		for i in range (len(donor_names)):
```

Comment:
better to put this in a separate function as well.

there should be as little logic code as possible in the interactive loop.